text/0134-remove-span-status.md

```diff
@@ -0,0 +1,89 @@
+# Remove Span.Status
+
+**Author:** Nikita Salnikov-Tarnovski, Splunk
+
+Remove `Span.status` and related APIs and conventions.
+
+## Motivation
+
+During numerous heated discussions in various pull requests (see
+<https://github.com/open-telemetry/opentelemetry-specification/issues/599>
+<https://github.com/open-telemetry/opentelemetry-specification/pull/697>
+<https://github.com/open-telemetry/opentelemetry-specification/pull/427>
+<https://github.com/open-telemetry/oteps/pull/69>
+<https://github.com/open-telemetry/oteps/pull/86>
+<https://github.com/open-telemetry/oteps/pull/123>)
+and during Errors Working Group meetings a lot of concerns were raised about suitability of the current `Span.Status`
+field encoded as Google RPC code for recording a result of arbitrary monitored operations from various domains
+(http requests, database queries, file system operations etc). It is hard to define clear and umambigous mapping
+from those domain to a fixed enum originated from RPC.
+
+Therefore the proposal is to remove the `Span.Status` field from Trace API and OTLP.
+
+## Internal details
+
+Several areas are affected by this change and will require a coordinated effort to make this happen.
+
+### Trace API Specification
+
+* Remove all mentions of `Status` and `Set Status` API from Span section.
+* Remove whole `Status` section.
+
+### Trace SDK Specification
+
+No changes needed.
+
+### Trace SDK exporters
+
+Remove translation rules pertaining to `Span.Status`.
+
+### Span semantic conventions
+
+* From HTTP conventions, remove `Status` section.
+* In gRPC conventions, replace the requirement to set span status with the requirement to set span attribute `grpc.status`
+to the canonical status code of the gRPC response.
+
+### OTLP data definition
+
+Rename `Span.status` field to `Span.deprecated_status` field.
+
+### Language API and SDK
+
+* Every language must remove status related code from their implementations of Trace API and SDK.
+* Every language must remove all code in their span exporters which serializes span status.
+* All instrumentation libraries, both manual and automated, must remove all calls to the removed `Span.setStatus` API.
+* All instrumentation libraries, both manual and automated, must adapt new semantic conventions for gRPC spans.
+
+### Collector
+
+Collector will have a transition period during which it has a special handling both for old data,
+which has `Span.Status` present, and new data without Status.
+This will allow for older pipelines, still relying on `Span.Status` to continue to work with as little disruption as possible.
+The exact duration of this transition period will be left for Collector's maintainers to decide.
+
+* If incoming OTLP data has `Span.Status` present, that will NOT be removed and will be passed on as-is.
+* If incoming OTLP data has `Span.Status` present and it is not equal to `OK`,
```

Comment:
Should a non-empty message also be translated independently of the status code?

Reply:
Message is part of the `Status`, how can it be independent?

Reply:
I mean, it makes sense to add deprecated_status when "incoming OTLP data has `Span.Status` present and it is not equal to `OK`", but it may make sense to independently set deprecated_message when "incoming OTLP data has `Span.Status` present and **message non-empty**`"

Reply:
I think having message without code will be confusing, because current state is exactly the opposite: you may have code without message.

Reply:
But you may also have a message with an OK code. If you are worried about confusion, then set the status code additionally when the message is non-empty.